packages/frontend-v2/pages/login.tsx

```diff
@@ -0,0 +1,135 @@
+import { useRouter } from "next/router";
+import { useState, useEffect } from "react";
+import { useForm } from "react-hook-form";
+import { LoginStudentDto } from "../temp/dto-types";
+import { AxiosError } from "axios";
+
+const Login: NextPage = () => {
+  const [apiError, setApiError] = useState("");
+  const [renderSpinner, setRenderSpinner] = useState(true);
+  const router = useRouter();
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors, isSubmitting },
+  } = useForm<LoginStudentDto>({
+    resolver: classValidatorResolver(LoginStudentDto),
+    mode: "onTouched",
+    shouldFocusError: true,
+  });
+
+  const loginWithCookie = async () => {
+    setRenderSpinner(true);
+    try {
+      const student = await API.student.getMe();
+      if (student) {
+        if (student.isOnboarded) {
+          // Redirect to home
+          router.push("/home");
+        } else {
+          // Redirect to onboarding
+          router.push("/onboarding");
+        }
+      }
+    } catch (err) {
+      const error = err as AxiosError;
+      console.log(error);
+      setRenderSpinner(false);
+    }
+  };
+
+  useEffect(() => {
+    loginWithCookie();
+  }, []);
+
+  const onSubmitHandler = async (payload: LoginStudentDto) => {
+    try {
+      const user = await API.auth.login(payload);
+      if (user) {
+        if (user.isOnboarded) {
+          // redirect home
+          console.log("redirect to home");
+          router.push("/home");
+        } else {
+          // redirect to onboarding
+          console.log("redirect to onboarding");
+          router.push("/onboarding");
+        }
+      }
+    } catch (err) {
+      const error = err as AxiosError;
+      if (error.response?.status === 401)
+        setApiError("Invalid credentials, please try again.");
+      else setApiError(error.message);
+      console.log(error);
+    }
+  };
+
+  if (renderSpinner)
+    return (
+      <Text pt="5%" fontSize="3xl" color="red.300">
+        Loading
+      </Text>
+    );
+
+  return (
+    <form onSubmit={handleSubmit(onSubmitHandler)}>
+      <h2>Log In</h2>
+      <br />
+
+      {apiError && (
+        <Text
+          pt="5%"
+          fontSize={{ desktop: "3xl", laptop: "2xl", tablet: "xl" }}
+          color="red.300"
+        >
+          {apiError}
+        </Text>
+      )}
+
+      <FormControl isInvalid={errors.email != null}>
+        <Input
+          id="email"
+          placeholder="example@email.com"
+          {...register("email")}
+        />
+        <FormErrorMessage>{errors.email?.message}</FormErrorMessage>
+      </FormControl>
+
+      <FormControl isInvalid={errors.password != null}>
+        <InputGroup>
+          <Input
+            type={"password"}
+            id="password"
+            placeholder="Enter Password"
+            {...register("password")}
+          />
+        </InputGroup>
+        <FormErrorMessage>{errors.password?.message}</FormErrorMessage>
```

Comment:
how can i manually test the frontend? i go to 127.0.0.1:3002 and can't seem to click the sign in button (it does nothing). then also visiting /login results in an error see belo
![Screen Shot 2022-07-29 at 8 53 53 PM](https://user-images.githubusercontent.com/22948432/181863990-21739ed1-7b61-4834-9561-cf4647393fe6.png)
w

Reply:
@BrandonLim8890 are you able to test it locally?

Reply:
Yea I can test locally. I'm not sure why it's giving you that error